pkg/api/customization/app/app.go

```diff
@@ -94,14 +94,22 @@ func (w Wrapper) ActionHandler(actionName string, action *types.Action, apiConte
 		return err
 	}
 
+	var appMap map[string]interface{}
+	if err := access.ByID(apiContext, &projectschema.Version, projectv3.AppType, apiContext.ID, &appMap); err != nil {
+		return httperror.NewAPIError(httperror.NotFound, fmt.Sprintf("unable to access app by id: %v", err))
+	}
+
+	if err := apiContext.AccessControl.CanDo(pv3.AppGroupVersionKind.Group, pv3.AppResource.Name, "update", apiContext, appMap, apiContext.Schema); err != nil {
+		return httperror.NewAPIError(httperror.PermissionDenied, fmt.Sprintf("user does not have permission to update for action %s", actionName))
+	}
+
 	creatorNotFound := false
```

Comment:
probably dont need this any more. just move it in to one if kicking the error now that the compound conditional is removed.

Reply:
reference in upgrade and rollback logic